tools/xhprof/docker-compose.override.yml

```diff
@@ -0,0 +1,137 @@
+version: '3.7'
+
+services:
+
+  ##
+  # The web server container.
+  ##
+  wordpress-develop:
+    image: nginx:alpine
+
+    networks:
+      - wpdevnet
+
+    ports:
+      - ${LOCAL_PORT-8889}:80
+
+    environment:
+      LOCAL_DIR: ${LOCAL_DIR-src}
+
+    volumes:
+      - ./tools/local-env/default.template:/etc/nginx/conf.d/default.template
+      - ./:/var/www
+
+    # Load our config file, substituting environment variables into the config.
+    command: /bin/sh -c "envsubst '$$LOCAL_DIR' < /etc/nginx/conf.d/default.template > /etc/nginx/conf.d/default.conf && exec nginx -g 'daemon off;'"
+
+    depends_on:
+      - php
+
+  ##
+  # The PHP container.
+  ##
+  php:
+    image: spacedmonkey/php-xhprof:${LOCAL_PHP-latest}
+    networks:
+      - wpdevnet
+
+    environment:
+      - LOCAL_PHP_XDEBUG=${LOCAL_PHP_XDEBUG-false}
+      - LOCAL_PHP_MEMCACHED=true
+      - PHP_FPM_UID=${PHP_FPM_UID-1000}
+      - PHP_FPM_GID=${PHP_FPM_GID-1000}
+
+    volumes:
+      - ./tools/local-env/php-config.ini:/usr/local/etc/php/conf.d/php-config.ini
+      - ./:/var/www
+
+    depends_on:
+      - mysql
+      - memcached
+      - xhgui
+
+  xhgui:
+    # https://hub.docker.com/r/spacedmonkey/xhgui/tags
+    image: spacedmonkey/xhgui:latest
+    restart: always
+    environment:
+      - XHGUI_MONGO_HOSTNAME=mongo
+      - XHGUI_MONGO_DATABASE=xhprof
+    ports:
+      - "8142:80"
```

Comment:
I think the ports you have here are fine, but it will cause conflicts if someone is trying to run both a profiler against a plugin and another one against core for some reason. To avoid conflicts we may want to consider using different ports for the Mongo DB and the XHGUI application.

Reply:
This can be changed to whatever you want. This is just example. Maybe once this is in core, we can use an env variable to make this configable.